docs/CONTRIBUTING.md

```diff
@@ -14,17 +14,31 @@ If you are a *git ninja*, we feel happy and sad for you at the same time. Your s
+* **First rule, be patient, be kind, rewind**. We're all volunteers with busy lives but usually very responsive in cataloging *pull requests* and pinging the relevant maintainer(s). Although not valid for the whole codebase, specific components of Kodi have a set of [maintainers/code owners](https://github.com/xbmc/xbmc/blob/master/.github/CODEOWNERS) whose review will be automatically requested when you open a pull request.
+* **Jenkins code formatting requests**. Always apply code style formatting requests from Jenkins. If not, please provide a valid reason why you do not apply them.
+* **Jenkins CI builds**. It is mandatory to have green light on all Jenkins building platforms. The "No-Jenkins" label can be set by maintainers to avoid having the CI building the PR code (and wasting build resources) in cases the contribution is still work in progress or if known in advance it won't build.
+* **Reviewing time**. Be patient, in general terms we speak of weeks not days, the active team is small and that means things can stack up, but  we are all doing our best. If your PR has been standing for too long without a review feel free to let us know.
+* **Reviewing process**. At least one member's approval is required to have green light to merge, to take into account that expertise is spread thinly across areas of Kodi, this makes getting reviews of some changes even harder. Kodi is tangled and what looks perfect to you may have unwanted side effects, so the approval by several members sometime is required, but in any case is always helpful, in particular for huge additions/refactors that may cause regressions. Therefore it is possible that your code will require more love, so more changes before its final version may be required, so arm yourself with relaxing herbal teas.
+* **Test builds**. When reasonable providing test builds to test your changes can speed up the review process, especially if you consider that some users or team members cannot build Kodi for testing. If you are a member of the kodi team, we remind you that there are jenkins helpers that can build and upload onto the mirrors where users can access them to test it. If you are not part of the team, you can make a request for test builds.
 
 ## Coding guidelines
 * **Follow our [code guidelines](CODE_GUIDELINES.md)**. New code should follow those guidelines even if existing code doesn't. If you're up to it, improve existing code and submit it in separate commits.
-* **Document the code**. You know you're brilliant, but maybe you'd like to understand what you did a year from now. Or, code God forbid, help others... Nope, `// Magic. Do not touch!` does not count as documentation.
+* **Document the code**. When changes are made to the code, its functionality is clear to the author, but unlike other users who read the block of code (even after several years) may not be facilitated. Providing documentation saves everyone the time spent studying a block of code over time.
```

Comment:
```suggestion
* **Document the code**. When changes are made to the code, its functionality may be clear to the author, but other developers who read the block of code (even after several years) can easily have a hard time understanding it. Providing documentation saves time for everyone.
```

as a side-note, after several years I'm quite sure that even being the original author of the code will not help you understand it any more :-)